src/main/java/org/ballerinax/datamapper/DataMapperPlugin.java

```diff
@@ -122,19 +121,13 @@ public void process(PackageNode packageNode) {
         }
 
         if (clientFlag) {
-            String pkgRoot = ((BLangPackage) packageNode).repos.resolve(((BLangPackage) packageNode).packageID).inputs.
-                    get(0).toString();
-            projectSourceFolder = pkgRoot.substring(0, pkgRoot.lastIndexOf("/"));
-            projectSourceFolder = projectSourceFolder.substring(0, projectSourceFolder.lastIndexOf("/"));
-            projectSourceFolder = projectSourceFolder.substring(0, projectSourceFolder.lastIndexOf("/"));
-
             for (TopLevelNode topLevelNode : ((BLangPackage) packageNode).topLevelNodes) {
                 if (topLevelNode.getKind() == NodeKind.CLASS_DEFN &&
                         (((BLangClassDefinition) topLevelNode).symbol.flags & Flags.CLIENT) == Flags.CLIENT) {
                     try {
                         extractFunctionCalls((BLangClassDefinition) topLevelNode);
                     } catch (Exception e) {
-                        dlog.logDiagnostic(DiagnosticSeverity.ERROR, packageNode.getPosition(), e.getMessage());
+                        dlog.error(packageNode.getPosition(), DiagnosticErrorCode.INCOMPATIBLE_TYPES, e.getMessage());
```

Comment:
Logging an internal error as an `INCOMPATIBLE_TYPES` compiler-error is not correct. Because internal errors are not something the user can fix.

If these are internal errors, the better approach would be to define a custom exception and throw it. Then the compiler will produce a `bad sad` error.

Applicable to all places.

Reply:
Fixed in [3e096a4](https://github.com/ballerina-platform/module-ballerinax-datamapper/pull/36/commits/3e096a4eb7a611475f39387c845e498230d53d7c)